apps/src/templates/sectionProgressV2/OpenAllStudentProgressButton.jsx

```diff
@@ -0,0 +1,149 @@
+import throttle from 'lodash/debounce';
+import PropTypes from 'prop-types';
+import React, {useState, useEffect, useRef} from 'react';
+import {connect} from 'react-redux';
+
+import PopUpMenu from '@cdo/apps/lib/ui/PopUpMenu';
+import {studentShape} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import i18n from '@cdo/locale';
+
+import FontAwesome from '../FontAwesome';
+import {
+  collapseMetadataForStudents,
+  expandMetadataForStudents,
+} from '../sectionProgress/sectionProgressRedux';
+
+import style from './expand-all-rows-dropdown.module.scss';
+
+const DROPDOWN_OFFSET = 172;
+
+function OpenAllStudentProgressButton({
+  students,
+  expandMetadataForStudents,
+  collapseMetadataForStudents,
+}) {
+  const [opened, setOpened] = useState(false);
+  const [menuLocation, setMenuLocation] = useState({menuTop: 0, menuLeft: 0});
+  const elementRef = useRef(null);
+  const resizeListener = useRef(null);
+
+  const getAllStudentIds = students.map(student => student.id);
+
+  const expandMetaDataForAllStudents = () => {
+    expandMetadataForStudents(getAllStudentIds);
+    setOpened(false);
+  };
+
+  const collapseMetaDataForAllStudents = () => {
+    collapseMetadataForStudents(getAllStudentIds);
+    setOpened(false);
+  };
+
+  const getMenuLocation = () => {
+    const rect = elementRef.current.firstChild.getBoundingClientRect();
+    console.log(
+      rect.bottom,
+      window.pageYOffset,
+      rect.right,
+      window.pageXOffset
+    );
+    return {
+      menuTop: rect.bottom + window.pageYOffset,
+      menuLeft: rect.right + window.pageXOffset - DROPDOWN_OFFSET,
+    };
+  };
+
+  const updateMenuLocation = throttle(() => {
+    setMenuLocation(getMenuLocation());
+  }, 50);
+
+  useEffect(() => {
+    if (opened && !resizeListener.current) {
+      resizeListener.current = updateMenuLocation;
+      window.addEventListener('resize', resizeListener.current);
+      setMenuLocation(getMenuLocation());
+    } else if (!opened && resizeListener.current) {
+      window.removeEventListener('resize', resizeListener.current);
+      resizeListener.current = null;
+    }
+
+    return () => {
+      if (resizeListener.current) {
+        window.removeEventListener('resize', resizeListener.current);
+      }
+    };
+  }, [updateMenuLocation, opened]);
+
+  const handleDropdownClick = () => {
+    setOpened(!opened);
+  };
+
+  const closeMenu = () => {
+    if (opened) setOpened(false);
```

Comment:
```suggestion
    setOpened(false);
```
It will be the same if we just call setOpened to false